flybywiresim/a32nx
# feat: EFB Colored Metar and Baro Setting

Fixes #6615 

## Summary of Changes
Adds coloring to raw Metar text based on the rules shown in the image below. 
Adds display of pressure according to the EFB Setting and airport location. 
Adds some improved Metar parsing and error handling. 

## Screenshots (if necessary)
Colored Metar
![image](https://user-images.githubusercontent.com/16833201/151707180-2ea8b313-34ab-4ebc-b683-6617bb0d6934.png)

Baro type according to airport (when EFB Setting is on AUTO)
![image](https://user-images.githubusercontent.com/16833201/151693495-3ea5adf3-96db-4c31-998c-43e6e7beb6da.png)
![image](https://user-images.githubusercontent.com/16833201/151693873-5be53c2e-d56b-4fe5-bc45-1231cd51ac87.png)

Reference doc:
![image](https://user-images.githubusercontent.com/16833201/151693747-91858d00-02bb-4b41-acce-c14d57eb0f5f.png)

Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions
- Check different Metars for correct coloring using the above coloring rules
  - Source for airports with bad weather for testing: https://www.badbadweather.com/
- Check if EFB Settings for Baro are respected for the Summary (icons) view of pressure:
  - AUTO: use airport ICAO to determine mb vs inHg
  - in Hg or hPa: Use the explicit setting


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
